.pre-commit-config.yaml

```diff
@@ -48,10 +48,23 @@ repos:
     rev: 4.0.1
     hooks:
       - id: flake8
+        args: ["--ignore=E501,W503,E203"]
         name: check PEP8
 
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.34.0
+    hooks:
+      - id: pyupgrade
+
+  - repo: https://github.com/hadialqattan/pycln
+    rev: v1.2.5
+    hooks:
+      - id: pycln
+        args: [--all, --expand-stars]
```

Comment:
NOTE: we have a small issue (sometimes annoying) with pycln. In this scenario
```python
from some_module import other_api_fun

def this_api_fun():
   return "hoi"


__all__ = ("other_api_fun", "this_api_fun" )
```
it will remove ``from some_module import other_api_fun`` because he is not able to associate the string in ``__all__`` with the function. 

[According to the manual](https://hadialqattan.github.io/pycln/#/?id=all-__all__) that should be handle differently but i guess we have something configured wrong with it.

My solution is to add an ``assert`` (which in production is removed since we use ``ENV PYTHONOPTIMIZE=TRUE``)

```python
from some_module import other_api_fun

def this_api_fun():
   return "hoi"

assert this_api_fun # nosec

__all__ = ("other_api_fun", "this_api_fun" )
```

Let me know if this happens to you as well. Perhaps we can find what is wrong with our config

Reply:
@pcrespov the problem is the '--all'. From the docs:
- Remove all unused import statements whether they has side effects or not!
- Faster results, because Pycln will skip side effects analyzing.

Reply:
RTFM @pcrespov :-D